### PR TITLE
Added updateWorkspace in workspace Controller

### PR DESCRIPTION
**Pull request by criticic**
Issue: #15 

- Added updateWorkspace function in workspace controller
- checks:
- - If any field is empty
- - if workspace exists
- - if workspace created by userId
- - if no other workspace has same workspaceId

### CODE REVIEW

**Comment by criticic** [comment]
I have added the verifications in the workspace controller itself, but it can also go to workspaceMiddleware or something (like the user controller). idk which is better choice

**Comment by automatefreely** [comment]
Don't use 500 status code for every error message. Read this article -> https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

**Comment by criticic** [comment]
I will fix it asap.

File: src/controllers/workspace.controller.js [hunk 1]
```diff
@@ -20,4 +20,43 @@ exports.createWorkspace = async (req, res) => {
...
+        if (workspace.createdBy !== userId) {
+            return response_500(res, "Invalid Request: User not authorized to update workspace")
+        }
+
+        const workspaceIDExists = await Workspace.findOne({workspaceId: workspaceId}).exec();
+        if (workspaceIDExists) {
+            return response_500(res, "Invalid Request: workspaceId already in use")
+        }
+
+        workspace = await Workspace.findByIdAndUpdate(req.params.id, {
```
Comment (automatefreely) [inline src/controllers/workspace.controller.js:46]: This is a error line: assignment to const variable.

**Comment by criticic** [comment]
@automatefreely lmk if it needs any other changes

**Pull request merged**